staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -119,6 +143,13 @@ type CustomResourceDefinitionStatus struct {
 	// AcceptedNames are the names that are actually being used to serve discovery
 	// They may be different than the names in spec.
 	AcceptedNames CustomResourceDefinitionNames `json:"acceptedNames" protobuf:"bytes,2,opt,name=acceptedNames"`
+
+	// StoredVersions are all versions ever marked as storage in spec. Tracking these
+	// versions allow a migration path for stored version in etcd. The field is mutable
+	// so the migration controller can first make sure a version is certified (i.e. all
```

Comment:
... can first finish a migration to another version (i.e. that no old objects are left in the storage), and then remove the rest of the versions from this list.